qiskit_research/utils/dynamical_decoupling.py

```diff
@@ -182,3 +192,51 @@ def add_pulse_calibrations(
                 pulse.play(inverted_pulse, x_instruction.channel)
             for circ in circuits:
                 circ.add_calibration("ym", [qubit], sched)
+
+
+def get_urdd_phis(urdd_pulse_num: int = 4) -> Sequence[float]:
+    """Gets \\phi_k values for n pulse UR sequence"""
+    if urdd_pulse_num % 2 == 1:
+        raise ValueError("urdd_pulse_num must be even")
+    if urdd_pulse_num < 4:
+        raise ValueError("urdd_pulse_num must be >= 4")
+
+    # get capital Phi value
+    if urdd_pulse_num % 4 == 0:
+        m_divisor = int(urdd_pulse_num / 4)
+        big_phi = np.pi / m_divisor
+    else:
+        m_divisor = int((urdd_pulse_num - 2) / 4)
+        big_phi = (2 * m_divisor * np.pi) / (2 * m_divisor + 1)
+
+    # keep track of unique phi added; we choose phi2 = big_phi by convention--
+    # only real requirement is (n * big_phi = 2pi * j for j int)
+    unique_phi = [0, big_phi]
+    # map each phi in [phis] to location (by index) of corresponding [unique_phi]
+    phi_indices = [0, 1]
+    # populate remaining phi values
+    for kk in range(3, urdd_pulse_num + 1):
+        phi_k = (kk * (kk - 1) * big_phi) / 2
+        # values only matter modulo 2 pi
+        phi_k = (phi_k) % (2 * np.pi)
+        if np.isclose(phi_k, 0):
+            phi_k = 0
+        elif np.isclose(phi_k, 2 * np.pi):
+            phi_k = 0
+
+        added_new = False
+        for idx, u_phi in enumerate(unique_phi):
+            if np.isclose(u_phi, phi_k, atol=0.001):
+                added_new = True
+                phi_indices.append(idx)
+
+        if added_new is False:
+            unique_phi.append(phi_k)
+            phi_indices.append(len(unique_phi) - 1)
+
+    # construct phi list
+    phis: List[float] = []
```

Comment:
Prefer using builtins for new type annotations: `list[float]`.

Reply:
Changed!